apps/reducers/lib/services/domoget.ex

```diff
@@ -53,6 +56,174 @@ defmodule Service.Domo do
+    end
+
+
+
+
+    def sort_by_time_period(filtered_meta, date_field) do
+      time_periods = Application.get_env(:reducers, Application.get_env(:reducers, :current_periods))
+      filtered_meta
+      |> Enum.group_by(fn(row)-> find_period(row, time_periods, date_field) end)
+    end
+
+    def find_period(row, season_periods, date_field) do
+      timestamp = date(row[date_field])
+      {period, _data} = season_periods
+      |>  Enum.find({:out_of_season, "data"}, fn {_period, %{:start_time => start_time, :end_time => end_time}} -> timestamp >= start_time and timestamp <= end_time end)
+      to_string(period)
+    end
+
+
+  def get_timestamp({:bin_audit, event}), do: date(event.data["DATE"])
+  def get_timestamp({:actuals, event}), do: date(event.data["Week"])
+  def get_timestamp({:pre_actual, event}), do: date(event.data["Week"])
+  def get_timestamp({:refill_actual, event}), do: date(event.data["Week"])
+  def get_timestamp({:pre_challenge, event})do
+    max = Enum.max_by(event.data["users"], fn(user) -> elem(user, 1)["start_time"] end)
+    elem(max, 1)["start_time"]
+  end
+  def get_timestamp({:refill_challenge, event})do
+    max = Enum.max_by(event.data["users"], fn(user) -> elem(user, 1)["start_time"] end)
+    elem(max, 1)["start_time"]
+  end
+
+  def date(datestring) when datestring == "", do: 1494287000000 #this date will be before season1 starts, so it won't get played
+  def date(datestring) when is_nil(datestring), do: 1494287000000 #this date will be before season1 starts, so it won't get played
+  def date(datestring) when datestring == "0", do: 1494287000000 #this date will be before season1 starts, so it won't get played
+  def date(datestring) do
+    {year, month, day} = case Regex.match?(~r/-/, datestring) do
+      true -> [year, month, day] = String.split(datestring, "-")
+      {year, month, day}
+      false ->[month, day, year] = String.split(datestring, "/")
+      {year, month, day}
+    end
+    seconds = Timex.to_datetime({{String.to_integer(year), String.to_integer(month), String.to_integer(day)}, {16, 0, 0}})
+    |> Timex.to_unix()
+    seconds * 1000
+  end
+
+
+
+  def merge_two_stores(dataset, _type, store_1, store_2) when is_list(dataset) do
+    {store_1, store_1_data} = List.keyfind(dataset, store_1, 0)
+    {store_2 ,store_2_data} = List.keyfind(dataset, store_2, 0)
+
+    new_store_data = {store_1, store_1_data ++ store_2_data}
+
+    new_dataset = List.keydelete(dataset, store_1, 0)
+    |> List.keydelete(store_2, 0)
+
+    [new_store_data | new_dataset]
+  end
+
+  def merge_two_stores(dataset, type,  store_1, store_2) do
+    case type do
+      "bin_audit" -> dataset
+      _ ->
+        store_1_data = dataset[store_1]
+        store_2_data = dataset[store_2]
+
+        Map.drop(dataset, [store_1, store_2])
+        |> Map.put(store_1, store_1_data ++ store_2_data )
+    end
+  end
+
+  def merge_three_stores(dataset, _type, store_1, store_2, store_3) when is_list(dataset) do
```

Comment:
Could merge two & three stores be generically changed to just be "merge_stores"?

(When merging three stores (A,B, and C), could we make it simply two operations of merge stores?

e.g.
X = merge(A,B)
Y = merge(X, C)